tests/unit/lms/services/oauth_http_test.py

```diff
@@ -225,6 +231,51 @@ def test_refresh_access_token_raises_ValidationError_if_the_response_is_invalid(
                 sentinel.token_url, sentinel.redirect_uri, sentinel.auth
             )
 
+    @pytest.mark.parametrize("prevent_concurrent_refreshes", (False, True))
+    def test_refresh_access_token_acquires_refresh_lock(
+        self, svc, oauth2_token_service, prevent_concurrent_refreshes
+    ):
+        svc.refresh_access_token(
+            sentinel.token_url,
+            sentinel.redirect_uri,
+            sentinel.auth,
+            prevent_concurrent_refreshes=prevent_concurrent_refreshes,
+        )
+        if prevent_concurrent_refreshes:
+            oauth2_token_service.try_lock_for_refresh.assert_called_once()
+        else:
+            oauth2_token_service.try_lock_for_refresh.assert_not_called()
```

Comment:
Similar comment here, I'd write two separate tests instead of one test with an `if`. One test for the default behaviour (no `prevent_concurrent_refreshes` argument passed), and one test for `prevent_concurrent_refreshes=True`. I prefer to avoid logic in test code.

This is sort of two separate tests masquerading as one. The test is called `test_refresh_access_token_acquires_refresh_lock()` but it actually asserts that it _does not_ acquire the lock in the case of `prevent_concurrent_refreshes=False`.

It just takes a bit of time for the eye to notice the `paramatrize()` and the `if` / `else` and to notice there are two tests here. You have to actually read the code to see it. Whereas if there are two separate tests it's more immediate to see, you just have to read the test names.